### PR TITLE
fix(neorg)!: fix breaking change in v8.0.0

### PR DESCRIPTION
**Pull request by ALVAROPING1**
## 📑 Description

Neorg recently released version 8.0.0 with a breaking change that requires slightly changing the config. This PR applies the necessary changes to get the plugin working correctly again. This is a breaking change because it requires a couple new system dependencies: Lua 5.1/LuaJIT and the Lua header files.

## ℹ Additional Information

Neorg now includes a `build.lua` file to handle the build process. The old build step specified with `build` is no longer sufficient, so specifying it leads to a broken installation as it overrides the included `build.lua` script.

Neorg now uses luarocks to manage its own dependencies, which is automatically compiled and installed by the `luarocks.nvim` plugin dependency using a system installation of Lua 5.1/LuaJIT with the Lua header files.

### CODE REVIEW

**Comment by Uzaaft** [comment]
See #634.

File: lua/astrocommunity/note-taking/neorg/init.lua [hunk 1]
```diff
@@ -1,23 +1,29 @@
...
-      ["core.journal"] = {}, -- Enables support for the journal module
-      ["core.dirman"] = { -- Manages Neorg workspaces
-        config = {
-          workspaces = {
-            notes = "~/projects/notes",
+  {
+    "vhyrro/luarocks.nvim",
+    priority = 1000, -- We'd like this plugin to load first out of the rest
+    config = true,
+  },
```
Comment (Uzaaft) [inline lua/astrocommunity/note-taking/neorg/init.lua:6]: This should. be a dependencies of neorg.

Comment (ALVAROPING1) [reply lua/astrocommunity/note-taking/neorg/init.lua]: fixed

**Comment by Uzaaft** [comment]
Changed target branch to v4. 
@ALVAROPING1  Are you running v4 or v3 of astronvim?

**Comment by ALVAROPING1** [comment]
> Changed target branch to v4. @ALVAROPING1 Are you running v4 or v3 of astronvim?

I'm running v3 of astronvim currently

**Comment by Uzaaft** [comment]
> > Changed target branch to v4. @ALVAROPING1 Are you running v4 or v3 of astronvim?
> 
> I'm running v3 of astronvim currently. If this is going to be postponed until after v4 is released, it might be worth it to pin neorg's version to its last commit before the recent changes (`583a82e`) as a temp fix, since otherwise this spec won't work with neorg's latest commit

Sure, sounds like a plan. I'll do that in v4 then. The v3 branch is frozen until then

**Review by Uzaaft — APPROVED** [review]

File: lua/astrocommunity/note-taking/neorg/init.lua [hunk 2]
```diff
@@ -1,8 +1,12 @@
 return {
   "nvim-neorg/neorg",
```
Comment (Uzaaft) [inline lua/astrocommunity/note-taking/neorg/init.lua:2]:
```suggestion
  "nvim-neorg/neorg",
  version = "^8",
```

**Pull request merged**